src/Application/Processor/ProductLowestPriceBeforeDiscountProcessor.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\PriceHistoryPlugin\Application\Processor;
+
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingLogEntryInterface;
+use Sylius\PriceHistoryPlugin\Domain\Repository\ChannelPricingLogEntryRepositoryInterface;
+use Webmozart\Assert\Assert;
+
+final class ProductLowestPriceBeforeDiscountProcessor implements ProductLowestPriceBeforeDiscountProcessorInterface
+{
+    private const PERIOD_IN_DAYS = 30;
+
+    public function __construct(private ChannelPricingLogEntryRepositoryInterface $channelPricingLogEntryRepository)
+    {
+    }
+
+    public function process(ChannelPricingInterface $channelPricing): void
+    {
+        if (!$this->thereIsADiscount($channelPricing)) {
+            $channelPricing->setLowestPriceBeforeDiscount(null);
+
+            return;
+        }
+
+        $mostRecentLogEntry = $this->getMostRecentLogEntry($channelPricing);
```

Comment:
WDYT to keep the price change date in the ChannelPricing model? eg. `ChannelPricing::discountedAt`

Reply:
I meant, from a technical perspective, it's useful because you don't need to search for the most recent log entry.

Reply:
As we're logging DateTime, I was wondering if it's possible to have different date between `ChannelPricingChangeListener` and `EntityListener`.
Edit:
I made a test. Log entry snapshot time:
<img width="901" alt="image" src="https://user-images.githubusercontent.com/40125720/219439026-ac0bdaf4-e6d6-4ace-9afd-1498f686b550.png">
and LowestPriceProcessor (delayed with xdebug):
<img width="488" alt="image" src="https://user-images.githubusercontent.com/40125720/219439076-4706afe1-2e08-4e33-8d51-eb97211517e7.png">

Those listeners are called one after another, and micro time shifting is possible, so I think it's safer to simply fetch the latest one, wdyt?